tests/secrets/writeFile.test.ts

```diff
@@ -0,0 +1,136 @@
+import type { VaultName } from 'polykey/dist/vaults/types';
+import path from 'path';
+import fs from 'fs';
+import fc from 'fast-check';
+import { test } from '@fast-check/jest';
+import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
+import PolykeyAgent from 'polykey/dist/PolykeyAgent';
+import { vaultOps } from 'polykey/dist/vaults';
+import * as keysUtils from 'polykey/dist/keys/utils';
+import * as testUtils from '../utils';
+
+describe('commandWriteFile', () => {
+  const password = 'password';
+  const logger = new Logger('CLI Test', LogLevel.WARN, [new StreamHandler()]);
+  const stdinArb = fc.string({ minLength: 0, maxLength: 100 });
+  const contentArb = fc.constantFrom('content', '');
+  let dataDir: string;
+  let polykeyAgent: PolykeyAgent;
+  let command: Array<string>;
+  let vaultNumber: number = 0;
+
+  // Helper function to generate unique vault names
+  function genVaultName() {
+    vaultNumber++;
+    return `vault-${vaultNumber}` as VaultName;
+  }
+
+  beforeEach(async () => {
+    dataDir = await fs.promises.mkdtemp(
+      path.join(globalThis.tmpDir, 'polykey-test-'),
+    );
+    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
+      password,
+      options: {
+        nodePath: dataDir,
+        agentServiceHost: '127.0.0.1',
+        clientServiceHost: '127.0.0.1',
+        keys: {
+          passwordOpsLimit: keysUtils.passwordOpsLimits.min,
+          passwordMemLimit: keysUtils.passwordMemLimits.min,
+          strictMemoryLock: false,
+        },
+      },
+      logger: logger,
+    });
+  });
+  afterEach(async () => {
+    await polykeyAgent.stop();
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
+    });
+  });
+
+  test.prop([stdinArb, contentArb], { numRuns: 5 })(
```

Comment:
Using multiple runs on a test which uses RPC calls is very time-taking. Running five tests takes almost 10 seconds.

I'm not sure what to do with this. Is this a reasonable trade-off, or should I reduce the runs even more? Or this small number of runs cannot hope to catch any real bugs, so I need to run more tests?

Reply:
That shouldn't be the case. Can you flamegraph this?

Reply:
I'm not sure how to use flamegraph, so I have asked Brian's help with this.

In the meantime, I ran some more tests, and got an average of 400ms after trying to `cat` a secret. However, I got an average of 2789ms when I tried to run `secrets write` with the same string as what the file contained for the `secrets cat` tests (it being `"testing"`).

`secrets cat` relies on a Duplex Handler to stream file paths to the handler and stream file contents from the handler. However, `secrets write` relies on a Unary Handler to do so, as we want to upgrade to a Raw Handler, but it needs too much work for it to be included in the first iteration.

As such, I believe the performance impact is due to the different handlers. This is also reinforced by the fact that running a similar test on `secrets list` gets a similar average time of 519ms while it is using a Server Handler.

Reply:
I doubt it's the kind of handler that's the problem. Keep in mind that the main difference between list, cat and write is the fact that you're writing to a file this time. You wan't to sanity check that the writing itself isn't just this slow normally. Fortunately we can compare to the vaults test in `Polykey` to see how long a write takes there. Take that as the base line.

Reply:
We did some digging and narrowed it down to the middlewear. Possibly the password hashing step. Needs some more digging so we create a new issue to track it. https://github.com/MatrixAI/Polykey-CLI/issues/294